src/Confluent.Kafka/Consumer.cs

```diff
@@ -806,15 +807,16 @@ internal Consumer(ConsumerBuilder<TKey, TValue> builder)
             }
             else
             {
-                throw new ArgumentException("Invalid key deserializer configuration.");
+                // enforced by the builder class.
+                Trace.Fail("Both async and sync key deserializers were set.");
```

Comment:
`Trace.Fail` is semantically more appropriate for program logic checks (i.e. only occur if we made a mistake). If this ever happened, what would be best from a user experience? that's open for debate i think, i'm not 100% sure. Worth noting the behavior in the case of a `Trace.Fail` is customizable ( refer to https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.trace.fail?view=netframework-4.7.2 ), and it's built for this.

Reply:
note: the netframework case was causing test failures, so i reverted to Exception for this type of error.